src/components/text-editor/link-menu/limel-text-editor-link-menu.tsx

```diff
@@ -0,0 +1,146 @@
+import { Component, Prop, h, Event, EventEmitter, State } from '@stencil/core';
+import { EditorTextLink } from '../prosemirror-adapter/menu/types';
+import { Languages } from 'src/components/date-picker/date.types';
+import translate from '../../../global/translations';
+
+/**
+ * This component is a menu for editing a link in the text editor.
+ * It allows the user to input a title of a link and a link itself.
+ */
+
+@Component({
+    tag: 'limel-text-editor-link-menu',
+    shadow: true,
+    styleUrl: 'limel-text-editor-link-menu.scss',
+})
+export class TextEditorLinkMenu {
+    /**
+     * The link
+     */
+    @Prop()
+    public link: EditorTextLink;
+
+    /**
+     * Defines the language for translations.
+     * Will translate the translatable strings on the components.
+     */
+    @Prop({ reflect: true })
+    public language: Languages = 'en';
+
+    /**
+     * Open state of the dialog
+     */
+    @Prop()
+    public isOpen: boolean = false;
+
+    @State()
+    private invalidLink: boolean = false;
+
+    /**
+     * Emitted when the menu is closed from inside the component.
+     * (*Not* emitted when the consumer sets the `open`-property to `false`.)
+     */
+    @Event()
+    private cancel: EventEmitter<void>;
+
+    /**
+     * Emitted when the menu is saved from inside the component.
+     */
+    @Event()
+    private save: EventEmitter<void>;
+
+    /**
+     * Emitted when the user inputs new values for the link
+     */
+    @Event()
+    private linkChange: EventEmitter<EditorTextLink>;
```

Comment:
I guess this might be a quirk of prosemirror, and nothing we can do anything about, but I find it slightly weird that we emit updated values for the link as the user is typing, but do _not_ attach the current value to the `save` event.

Is that indeed due to prosemirror, or is it something we control, and could potentially change?